reconcile/gitlab_housekeeping.py

```diff
@@ -37,13 +41,26 @@ def clean_pipelines(dry_run, gl_project, pipelines, pipeline_timeout=60):
 
         # pipeline_timeout converted in seconds
         if elapsed > pipeline_timeout * 60:
-            logging.info(['canceling', p['web_url']])
-            if not dry_run:
-                try:
-                    gl_project.pipelines.get(p['id']).cancel()
-                except gitlab.exceptions.GitlabPipelineCancelError as err:
-                    logging.error('unable to cancel {} - error message {}'
-                                  .format(p['web_url'], err.error_message))
+            timeout_pipelines.append(p)
+
+    return timeout_pipelines
+
+
+def clean_pipelines(dry_run: bool, gl_instance: str, gl_project_id: int,
+                    gl_settings: str, pipelines: List[str]) -> None:
+    if not dry_run:
+        gl_piplelines = GitLabApi(gl_instance,
+                                  project_id=gl_project_id,
+                                  settings=gl_settings).project.pipelines
+
+    for p in pipelines:
+        logging.info(['canceling', p['web_url']])
+        if not dry_run:
+            try:
+                gl_piplelines.get(p['id']).cancel()
```

Comment:
What happens if the pipeline finishes or crashes in between the call to `get` and the cancel operation? Is it still going to raise a `GitlabPipelineCancelError`?

Reply:
Also, I wonder if it wouldn't be easier to just have one `dry_run` path. Say:

```python
if dry_run:
    logging.info(["cancelling", [p['web_url'] for p in pipeilnes])
    return

for p in pipelines:
   try:
     ...
...
```

Reply:
Won't this prevent logging when not in dry_run?

Reply:
> What happens if the pipeline finishes or crashes in between the call to `get` and the cancel operation? Is it still going to raise a `GitlabPipelineCancelError`?

No, it will return 200, but pipeline will not actually have any change.